internal/batcher/batcher.go

```diff
@@ -65,11 +65,11 @@ func New(itemExample interface{}, maxHandlers int, handler func(interface{}) err
 // If Shutdown has been called, Add immediately returns an error.
 func (b *Batcher) Add(ctx context.Context, item interface{}) error {
 	b.wg.Add(1)
```

Comment:
I think the `Add` needs to be protected by the mutex as well. `sync.WaitGroup` is not threadsafe.

Reply:
... apparently I'm wrong about that.